test/src/code_field/tabs_to_spaces_test.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_code_editor/src/code_field/text_editing_value.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+const _codeRowWithoutTabs = 'int a = 1;';
+const _codeWithTabs = '''
+class Foo {
+\tint field;
+\tvoid method() {
+\t\tprint(field);
+\t}
+}
+''';
+const _codeWithDoubleSpaces = '''
+class Foo {
+  int field;
+  void method() {
+    print(field);
+  }
+}
+''';
+const _spaceCount = 2;
+
+
+void main() {
+  group('Replace tabs with spaces', () {
+    test('Empty without selection', () {
+      TextEditingValue value = TextEditingValue.empty;
+      value = value.tabsToSpaces(_spaceCount);
+      
+      expect(value.text, '');
+      expect(value.selection.baseOffset, -1);
+      expect(value.selection.extentOffset, -1);
+    });
+
+    test('Empty with cursor at start', () {
+      TextEditingValue value = TextEditingValue.empty.copyWith(
+        selection: const TextSelection.collapsed(offset: 0),
+      );
+
+      value = value.tabsToSpaces(_spaceCount);
+      
+      expect(value.text, '');
+      expect(value.selection.baseOffset, 0);
+      expect(value.selection.extentOffset, 0);
```

Comment:
You can compare the whole selection to `TextSelection.collapsed(offset: 0)`
Here and below.